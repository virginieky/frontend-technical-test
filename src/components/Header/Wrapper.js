import styled from 'styled-components';

const Wrapper = styled.div` 
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 3;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 70px;
  background-color: white;
  border-bottom: 1px solid #e9e9e9;
  div {
    position: relative;
    width: 100%;
    height: 100%;
  }
`;

export default Wrapper;
