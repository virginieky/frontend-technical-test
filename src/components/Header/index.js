import React from 'react';
import { Container } from 'reactstrap';
import Image from 'next/image';

import Logo from '../../assets/lbc-logo.webp';

import Flex from '../Flex';
import Padded from '../Padded';
import Wrapper from './Wrapper';

const Header = ({
}) => {
  return (
    <Wrapper>
      <Container>
        <Flex alignItems="center" justifyContent="center" >
          <Padded top bottom>
            <div>
              <Image 
                src={Logo} 
                alt="Leboncoin's logo" 
                layout='fill'
                objectFit='contain' />
            </div>
          </Padded>
        </Flex>
      </Container>
    </Wrapper>
  );
};

export default Header;
        