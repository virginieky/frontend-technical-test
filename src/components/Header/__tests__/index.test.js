import React from 'react';
import { render } from '@testing-library/react';
import Header from '../index';

describe('Header', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});
