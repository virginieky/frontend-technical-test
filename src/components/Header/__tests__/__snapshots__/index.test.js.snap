// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Header should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Wrapper-sc-omo8oc-0 cxzqom"
  >
    <div
      class="container"
    >
      <div
        class="Flex-sc-1vrf9e3-0 kMTbkk"
      >
        <div
          class="Padded-sc-q6j7xl-0 dlzNrg"
        >
          <div>
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="Leboncoin's logo"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
